src/components/resumption/test/last_state_test.cc

```diff
@@ -64,55 +76,78 @@ class LastStateTest : public ::testing::Test {
   const std::string empty_dictionary_;
   const std::string app_info_dat_file_;
 
-  resumption::LastState last_state_;
+  ::utils::SharedPtr<resumption::LastState> last_state_;
+
+  void CompactJson(std::string& str) {
+    Json::Value root;
+    Json::Reader reader;
+    reader.parse(str, root);
+    Json::FastWriter writer;
+    str = writer.write(root);
+    if (str[str.size() - 1] == '\n') {
+      str.erase(str.size() - 1, 1);
```

Comment:
@AByzhynar , that is ok.  already checked that I cann't do in such way as you wrote